aspnet/fundamentals/localization.rst

```diff
@@ -1,8 +1,302 @@
+- ``AddViewLocalization`` Adds support for localized view files. In this sample view localization is based on the view file suffix. For example "fr" in the *Index.fr.cshtml* file.
+- ``AddDataAnnotationsLocalization`` Adds support for localized ``DataAnnotations`` validation messages through ``IStringLocalizer`` abstractions.
+
+
+Resource file naming
+---------------------
+
+When you create a new project with Visual Studio, the project name is used for the assembly name, project name and default namespace. For example, if you created a project named **X.Y.Z**, the assembly name, project name and default namespace would all be "X.Y.Z".
+
+.. image:: localization/_static/newproj.png
+
+When you create a type in a subfolder, Visual Studio appends the folder name to the default namespace (it nests the namespace). C# uses the "." between nested namespaces and types. Using the **X.Y.Z** project, a class named ``Cat`` in the *Models* folder would have the namespace ``X.Y.Z.Models``:
+
+.. code-block:: c# 
+
+    namespace X.Y.Z.Models
+    {
+        class Cat
+        {
+            public string name { get; set; }
+        }
+    }
+
+If you reference a type with the fully qualified namespace (longer than the default namespace), Visual Studio will warn you to use the nested or *simplified namespace*:
+
+.. image:: localization/_static/simp2.png
+
+In the image above, Visual Studio is warning you that ``Models.Cat`` can be used to for the type, because ``Models.Cat`` is in scope. The code below shows the ``Cat`` class with both the full namespace and the nested namespace.
+
+.. code-block:: c# 
+
+    namespace X.Y.Z
+    {
+        class Program
+        {
+            static void Main(string[] args)
+            {
+                X.Y.Z.Models.Cat cat2 = new X.Y.Z.Models.Cat();  // Full NS
+                Models.Cat cat = new Models.Cat();               // Simplified NS
+                cat.name = "spot";
+                System.Console.WriteLine(cat.name);
+            }
+        }
+    }
+
+When you create resource files based on a type, the resource file path must include the type with the simplified namespace. In the sample project, the default namespace is ``Localization.StarterWeb``. The ``HomeController`` reference with a simplified namespace is ``Controllers.HomeController``, therefore the French resource file must be named *Controllers.HomeController.fr.resx*. The ``ConfigureServices`` method sets the ``ResourcesPath`` to "Resources", so the project relative path for the home controllers French resource file is *Resources/Controllers.HomeController.fr.resx*. Alternatively, you can use folders to organize resource files. For the home controller, the path would be *Resources/Controllers/HomeController.fr.resx*. If you don't use the ``ResourcesPath`` option, the *.resx* file would go in the  project base directory. The resource file for ``HomeController`` would be named *Controllers.HomeController.fr.resx*.
```

Comment:
`the home controllers French resource file` -> `the home controller's French resource file`
`would go in the project base` (double space between `the` and `project`)